src/geometry.cpp

```diff
@@ -369,48 +366,47 @@ BoundaryInfo distance_to_boundary(Particle& p)
   double d_lat = INFINITY;
   double d_surf = INFINITY;
   int32_t level_surf_cross;
-  std::array<int, 3> level_lat_trans {};
+  array<int, 3> level_lat_trans {};
 
   // Loop over each coordinate level.
-  for (int i = 0; i < p.n_coord_; i++) {
-    const auto& coord {p.coord_[i]};
-    Position r {coord.r};
-    Direction u {coord.u};
+  for (int i = 0; i < p.n_coord(); i++) {
+    const auto& coord {p.coord(i)};
+    const Position& r {coord.r};
```

Comment:
Ah yeah... my thinking here is that we could possibly reduce register pressure by doing const references rather than keeping separate copied values. With a const reference, the compiler should be able to choose between storing a const pointer and keeping a copied value.

Reply:
Makes sense! I'm not sure if this would really make any difference in practice after compiler optimization, but definitely doesn't hurt anything either. For future refactoring PRs, might be best to cull changes like this that are independent from the PR topic and put into a separate optimization PR.